### DISCUSSION TITLE
Drush10 autogenerated aliases

### DISCUSSION THREAD

**Discussion by AlexSkrypnyk** [amazeeio/lagoon#2402] [General]
For Drush 8 we have been using an `aliases.drushrc.php` file like this:

```
<?php

// Don't change anything here, it's magic!
// @codingStandardsIgnoreFile
global $aliases_stub;
if (empty($aliases_stub)) {
  $ch = curl_init();
  curl_setopt($ch, CURLOPT_AUTOREFERER, TRUE);
  curl_setopt($ch, CURLOPT_HEADER, 0);
  curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
  curl_setopt($ch, CURLOPT_URL, 'https://drush-alias.lagoon.amazeeio.cloud/aliases.drushrc.php.stub');
  curl_setopt($ch, CURLOPT_FOLLOWLOCATION, TRUE);
  $aliases_stub = curl_exec($ch);
  curl_close($ch);
}
eval($aliases_stub);
```

Drush9+ have moved to using YAML-based configuration files and we have lost the ability to execute this script.
There is an [article](https://docs.lagoon.sh/lagoon/drupal/drush-9) that provides a workaround for this.

Drush10 provides command alteration that would probably allow to incorporate that stub from above.

https://raw.githubusercontent.com/drush-ops/drush/10.x/examples/Commands/SiteAliasAlterCommands.php

It would be very helpful if https://github.com/amazeeio/drupal-example-simple would contain a working example of such integration.

**Comment by luigimannoni** [comment]
The stub file is broken on Drush 11 and onwards as the `DRUSH_VERSION` constant is now a string, so the workaround does not work.
